### ISSUE TITLE
Add table azure_reserved_instance_recomendations

### ISSUE THREAD

**Issue by Lachlan-White** [turbot/steampipe-plugin-azure#666]
**References**
List API:
https://github.com/Azure/azure-sdk-for-go/blob/v58.0.0/services/consumption/mgmt/2019-10-01/consumption/reservationrecommendations.go#L49

**Comment by ParthaI** [comment]
@Lachlan-White, my apologies for not communicating on this issue for quite some time. I've created a [draft pull request](https://github.com/turbot/steampipe-plugin-azure/pull/696) addressing this matter, which includes the addition of a table called `azure_reservation_recommendation`. 

I can't test the code changes due to the lack of resource availability in the portal.

Would you be able to test it out on the PR branch `issue-666` and share your feedback with us regarding whether the new table fulfills the requirements?

Steps for testing out the branch `issue-666`:

1. Clone the repository: git clone https://github.com/turbot/steampipe-plugin-azure.git
2. Navigate to the steampipe-plugin-azure directory: cd steampipe-plugin-azure
3. Switch to the issue-637 branch: git checkout issue-666
4. Run the command `make` to build the changes.
5. Run the query(select * from azure_reservation_recommendation)

Note: It's possible to tailor the recommendations by applying filters on columns such as `look_back_period`, `resource_type`, or `scope` in the query parameters. By default, the table retrieves data for the resource type `VirtualMachines` and uses a `Single` scope, focusing on the usage of the past seven days to generate recommendations.

Thanks!

**Comment by Lachlan-White** [comment]
@ParthaI, no worries appreciate the effort.

I followed your steps and received the below output from the table:
![image](https://github.com/turbot/steampipe-plugin-azure/assets/41881046/d25bc858-b2a7-4bb2-93dd-61f876af5aef)

Expected output returning for the below though:
<img width="821" alt="image" src="https://github.com/turbot/steampipe-plugin-azure/assets/41881046/22c4ebdd-5bd3-4861-829a-7d0ed496adc4">

**Comment by ParthaI** [comment]
@Lachlan-White, Thanks for testing the PR branch and sharing the feedback with us.

I have been experimenting with APIs to enumerate all reservation recommendations for a specific scope.

According to the [Official API Documentation](https://learn.microsoft.com/en-us/rest/api/consumption/reservation-recommendations/list?view=rest-consumption-2023-05-01&tabs=HTTP&tryIt=true&source=docs#code-try-0), it is possible to filter and list reservation recommendations by specifying a scope. However, I've encountered a challenge where the API consistently yields an empty array, instead of the expected result sets.

I have initiated a support request with the Azure SDK repository to address and resolve this anomaly. For updates and further details on the progress of this issue, it can be tracked here: [Azure SDK GitHub Issue #22071](https://github.com/Azure/azure-sdk-for-go/issues/22071).

Thanks!
